### ISSUE TITLE
maspit does not handle '.,.'

### ISSUE THREAD

**Issue by knausb** [knausb/vcfR#53]
```
library(vcfR)
data("vcfR_test")
gt <- extract.gt(vcfR_test, element = 'HQ')
myHQ1 <- masplit(gt)
Failed to convert to a float.
Failed to convert to a float.
```

Needs to be fixed.

**Comment by knausb** [comment]
The internal function `str_vec_to_float_vec()` uses `std::vector<float>` as a return value. A side effect of this is that when missing data ('.') are encountered it can not be handled elegantly. A solution would be to rewrite `str_vec_to_float_vec()` to `str_vec_to_NumericVector`. `Rcpp::NumericVector` includes `NA_REAL` to encode missing data.

**Comment by knausb** [comment]
Using `Rcpp::NumericVector` would affect our downstream use of `std::sort`. Perhaps encoding missing data as -999 and then converting it to `NA_REAL` when inserting into the return matrix would be a better solution?

**Comment by knausb** [comment]
Fixed in vcfR 1.4.0.9000.
